### PR TITLE
Fix float 16 type support for some CUDA kernels

### PR DESCRIPTION
**Pull request by HectorSVC**
1. Fix the CreateConstantOnes for float16 type
2. Add cuda kernel code in the BatchNorm for float 16 type, there's issue to run cudnnBatchNormalizationForwardInference with float 16 type
3. Add float 16 test case for Gemm & BatchNorm CUDA kernel only

### CODE REVIEW

File: onnxruntime/core/providers/cuda/nn/batch_norm.cc [hunk 1]
```diff
@@ -46,6 +48,22 @@ Status BatchNorm<T>::ComputeInternal(OpKernelContext* p_op_kernel_context) const
   auto mean_data = reinterpret_cast<const CudaT*>(mean->template Data<T>());
   auto var_data = reinterpret_cast<const CudaT*>(var->template Data<T>());
 
+  if (X->DataType() == DataTypeImpl::GetType<MLFloat16>()) {
```
Comment (KeDengMS) [inline onnxruntime/core/providers/cuda/nn/batch_norm.cc]: Run all BN types through customized kernel? Seems no need to go through cudnn below

Comment (HectorSVC) [reply onnxruntime/core/providers/cuda/nn/batch_norm.cc]: cudnn one is faster for float type, will switch back to cudnn lib once fix the issue using cudnn.

**Comment by snnn** [comment]
Have you tried the 3 fp16 models?

File: onnxruntime/test/providers/cpu/nn/pool_op_test.cc [hunk 2]
```diff
@@ -52,6 +52,60 @@ TEST(PoolTest, MaxPool) {
   test.Run();
 }
 
+// Only CUDA kernel has float 16 support
+// Disable for now, still investigating the issue with cudnn lib
+//#ifdef USE_CUDA
+//TEST(PoolTest, MaxPool_F16) {
```
[the remark this answers is not in the document]
Comment (HectorSVC) [reply onnxruntime/test/providers/cpu/nn/pool_op_test.cc]: added back

---
In reply to: [253711064](https://github.com/Microsoft/onnxruntime/pull/436#discussion_r253711064) [](ancestors = 253711064)

**Comment by HectorSVC** [comment]
need to adjust the threshold for cuda test

---
In reply to: [460475067](https://github.com/Microsoft/onnxruntime/pull/436#issuecomment-460475067) [](ancestors = 460475067)

**Comment by snnn** [comment]
> need to adjust the threshold for cuda test, and a test failure on Linux. Will add them back after the test data updated.
> 
> In reply to: [460475067](https://github.com/Microsoft/onnxruntime/pull/436#issuecomment-460475067) [](ancestors = 460475067)

You may merge it, let me handle the other things at a later time.
